src/proc/namer.rs

```diff
@@ -86,6 +86,18 @@ impl Namer {
         })
     }
 
+    fn namespace(&mut self, reserved_keywords: &[&str], f: impl FnOnce(&mut Self)) {
+        let parent_unique = std::mem::take(&mut self.unique);
+        self.unique.extend(
```

Comment:
This is a bit unfortunate, it introduces a whole lot of heap allocations on every switch.
We need a more lightweight way to switch the namespaces